src/main/java/crazypants/enderio/machine/reservoir/BlockReservoir.java

```diff
@@ -69,11 +70,18 @@ public boolean onBlockActivated(World world, int x, int y, int z, EntityPlayer e
         }
         return true;
       }
+      if (FluidUtil.fillInternalTankFromPlayerHandItem(world, x, y, z, entityPlayer, tank)) {
```

Comment:
yes, as I wrote, you did that in a way I did not see.

And btw, your code leaves setDirty() hanging in the air now.